source/common/ssl/context_impl.cc

```diff
@@ -229,6 +229,14 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const ContextConfig& config, TimeS
 
     bssl::UniquePtr<EVP_PKEY> public_key(X509_get_pubkey(ctx.cert_chain_.get()));
     ctx.is_ecdsa_ = EVP_PKEY_id(public_key.get()) == EVP_PKEY_EC;
+    if (ctx.is_ecdsa_) {
+      // We only support P-256 ECDSA today.
+      EC_KEY* ecdsa_public_key = EVP_PKEY_get0_EC_KEY(public_key.get());
+      const int nid = EC_GROUP_get_curve_name(EC_KEY_get0_group(ecdsa_public_key));
+      if (nid != NID_X9_62_prime256v1) {
+        throw EnvoyException(fmt::format("ECDSA key is not P256 (NID = {})", nid));
```

Comment:
Can you match the text with other errors here, e.g.
```
fmt::format("Failed to load certificate chain from {}, only P-256 ECDSA certificates are supported", ctx.cert_chain_file_path_)
```

Reply:
Sure. Is the PR good on other fronts? If so, I can update the error message and add tests.

Reply:
Maybe check if `EVP_PKEY_get0_EC_KEY()` and `EC_KEY_get0_group()` return something, i.e.
```
EC_KEY* ecdsa_public_key = EVP_PKEY_get0_EC_KEY(public_key.get());
if (ecdsa_public_key == nullptr) {
    throw EnvoyException(...);
}
const EC_GROUP *ecdsa_group = EC_KEY_get0_group(ecdsa_public_key);
if (group == nullptr || EC_GROUP_get_curve_name(group) != NID_X9_62_prime256v1) {
    throw EnvoyException(...);
}
```
Throwing exceptions instead of `RELEASE_ASSERT()` to avoid malformed certificates sent via xDS from crashing Envoy at runtime.